packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -51,9 +53,14 @@ export const packageLayer: Packager = async (context, resource) => {
   if (packageHash) {
     await zipPackage(packageResult.zipEntries, destination);
   }
-  const zipFilename = packageHash
-    ? `${resource.resourceName}-${packageHash}-build.zip`
-    : resource.distZipFilename ?? `${resource.category}-${resource.resourceName}-build.zip`;
+
+  const layerCloudState = LayerCloudState.getInstance();
+  if (!layerCloudState.latestVersionLogicalId) {
+    // "Should" never be reachable, but sanity check just in case
+    throw new Error('LogicalId missing for new layer version.');
+  }
+
+  const zipFilename = `${resource.resourceName}-${layerCloudState.latestVersionLogicalId}-build.zip`;
```

Comment:
As this is a 2nd instance of generating a zip filename, could we put it into a function to reduce that cognitive load to have it maintained at 2 places if we are changing it?